src/prerender_content.js

```diff
@@ -27,30 +28,52 @@ const SITEMAP_OBJ_CLASSES_WHITELIST = [
   "Page",
 ];
 
+const SOURCE_DIR = "build";
+const TARGET_DIR = "buildPrerendered";
+
 async function prerenderContent() {
-  await prerenderSitemap(SITEMAP_OBJ_CLASSES_WHITELIST, window.storeResult);
-  await prerenderObjs(
-    PRERENDER_OBJ_CLASSES_BLACKLIST,
-    window.storeResult,
-    window.reportError
+  console.time("[prerenderContent]");
+
+  if (!process.env.SCRIVITO_ORIGIN) {
+    throw new Error(
+      'The environment variable "SCRIVITO_ORIGIN" is not defined.' +
+        " Prerendered pages need a configured origin or base URL."
+    );
+  }
+
+  const assetManifest = await fse.readJson(
+    path.join(SOURCE_DIR, "asset-manifest.json")
   );
-}
 
-// The following method will be overwritten by puppeteer in storePrerenderedContent.
-// It is only here, to simplify debugging in the browser
-window.storeResult = async ({ filename, content }) => {
-  console.log(
-    `[storeResult] received "${filename}" (file size: ${filesize(
-      content.length
-    )})`
+  console.log(`Removing ${TARGET_DIR}/`);
+  await fse.remove(TARGET_DIR);
+
+  console.log(`Copying ${SOURCE_DIR}/ to ${TARGET_DIR}/`);
+  await fse.copy(SOURCE_DIR, TARGET_DIR);
+
+  await fse.remove(path.join(TARGET_DIR, "asset-manifest.json"));
+
+  const sitemapFile = await prerenderSitemap(
+    TARGET_DIR,
+    SITEMAP_OBJ_CLASSES_WHITELIST
   );
-};
 
-// The following method will be overwritten by puppeteer in storePrerenderedContent.
-// It is only here, to simplify debugging in the browser
-window.reportError = (message, ...args) => {
-  console.log(`[reportError] ${message}`, ...args);
-};
+  const objFiles = await prerenderObjs(
+    TARGET_DIR,
+    PRERENDER_OBJ_CLASSES_BLACKLIST,
+    assetManifest
+  );
+
+  const storedFiles = [sitemapFile, ...objFiles].filter((f) => f !== undefined);
+
+  await extendRedirects(TARGET_DIR, storedFiles, SOURCE_DIR);
```

Comment:
Now that I see this code flow: Do we actually need to `extendRedirects` for the `sitemapFile`?

Also: How can `sitemapFile` be undefined, or `objFiles` contain undefined entries?

Reply:
Here https://github.com/Scrivito/scrivito_example_app_js/pull/387/files#diff-09630bc3ab6f736cd78af8b0863472db3ddc6327c6a150f86b2e4d76706da144R22 we can get `undefined` from `storeResult` in the EEXIST case https://github.com/Scrivito/scrivito_example_app_js/pull/387/files#diff-4d17c237861dbcf7ae91a039756943f9773d38974124980e14d88530b27b661cR22.

Reply:
> Now that I see this code flow: Do we actually need to extendRedirects for the sitemapFile

Not really. I just mindlessly recreated the old behavior.